# Explore Grid Gross Loss calculation

In [24]:
import pandas as pd

%reload_ext autoreload
%autoreload 2


# Tell python where to look for modules. 
import sys
sys.path.append('../../hourly-egrid/')

import src.download_data as download_data
import src.grid_gross_loss as ggl

In [26]:
ba_disposition = ggl.calculate_ba_gross_loss(year)

   ba_code  sales_for_resale_mwh  total_energy_losses_mwh  \
38    OVEC          9.033056e+06                 137962.0   
65     NaN          1.318416e+09                9278264.0   

    total_disposition_mwh  grid_gross_loss  
38           9.280962e+06         0.556509  
65           1.327948e+09         0.973410  


In [27]:
ba_disposition.sort_values(by="grid_gross_loss")

ba_code  sales_for_resale_mwh  total_energy_losses_mwh  \
37    NYMS          0.000000e+00             0.000000e+00   
30    MEMS          0.000000e+00             0.000000e+00   
44    PRMS          0.000000e+00             0.000000e+00   
3     AMPL          1.012920e+05             0.000000e+00   
32    NBSO          6.251665e+04             4.744000e+03   
..     ...                   ...                      ...   
19    GCPD          5.324303e+06             9.663700e+05   
35    NWMT          1.217508e+08             1.657638e+06   
10    CISO          2.753762e+08             4.371709e+07   
38    OVEC          9.033056e+06             1.379620e+05   
65     NaN          1.318416e+09             9.278264e+06   

    total_disposition_mwh  grid_gross_loss  
37           0.000000e+00         0.000000  
30           0.000000e+00         0.000000  
44           0.000000e+00         0.000000  
3            8.341200e+05         0.000000  
32           7.967359e+05         0.006461  
..                    ...              ...  
19           1.228537e+07         0.138825  
35           1.319013e+08         0.163307  
10           5.253259e+08         0.174904  
38           9.280962e+06         0.556509  
65           1.327948e+09         0.973410  

[66 rows x 5 columns]

In [35]:
ba_disposition.head(30)

ba_code  sales_for_resale_mwh  total_energy_losses_mwh  \
0      AEC          3.520000e+02             3.282280e+05   
1     AECI          9.842500e+04             7.852960e+05   
2     AKMS          5.456000e+04             1.565870e+05   
3     AMPL          1.012920e+05             0.000000e+00   
4      AVA          3.280487e+06             5.003510e+05   
5     AZPS          3.088308e+06             1.817961e+06   
6     BANC          3.439704e+06             7.540360e+05   
7     BPAT          9.453356e+07             1.715453e+06   
8      CEA          6.418300e+04             1.005140e+05   
9     CHPD          1.085249e+07             2.141400e+04   
10    CISO          2.753762e+08             4.371709e+07   
11    CPLE          2.301456e+07             2.817466e+06   
12    DOPD          3.464577e+06             5.843800e+04   
13     DUK          9.227782e+06             5.956771e+06   
14     EPE          3.735986e+06             5.432363e+05   
15    ERCO          2.637507e+08             1.309243e+07   
16    FMPP          1.295937e+06             3.659140e+05   
17     FPC          3.019323e+06             2.648872e+06   
18     FPL          1.095867e+07             8.233620e+06   
19    GCPD          5.324303e+06             9.663700e+05   
20     GVL          2.149400e+05             5.927100e+04   
21    HECO          0.000000e+00             2.768880e+05   
22    HIMS          0.000000e+00             1.296720e+05   
23     HST          0.000000e+00             9.781000e+03   
24     IID          1.087220e+05             3.148260e+05   
25    IPCO          1.887139e+06             1.066344e+06   
26    ISNE          1.674723e+07             3.395033e+06   
27     JEA          6.835000e+03             4.113860e+05   
28    LDWP          1.465896e+06             2.642531e+06   
29    LGEE          2.753273e+06             1.718370e+06   

    total_disposition_mwh  grid_gross_loss  
0            8.170971e+06         0.040172  
1            1.604540e+07         0.049244  
2            3.605061e+06         0.044103  
3            8.341200e+05         0.000000  
4            1.392932e+07         0.046986  
5            3.637640e+07         0.054613  
6            1.924186e+07         0.047717  
7            1.472526e+08         0.032540  
8            1.382154e+06         0.076264  
9            1.283366e+07         0.010809  
10           5.253259e+08         0.174904  
11           8.518864e+07         0.045316  
12           5.219010e+06         0.033309  
13           1.096095e+08         0.059341  
14           1.240823e+07         0.062641  
15           6.344030e+08         0.035323  
16           1.625518e+07         0.024461  
17           4.737721e+07         0.059716  
18           1.387034e+08         0.064454  
19           1.228537e+07         0.138825  
20           2.065285e+06         0.032032  
21           6.651997e+06         0.041625  
22           2.500285e+06         0.051863  
23           5.984540e+05         0.016344  
24           3.961361e+06         0.081717  
25           1.787949e+07         0.066678  
26           1.268831e+08         0.030826  
27           1.314033e+07         0.031323  
28           2.732020e+07         0.102209  
29           3.489954e+07         0.053455

# Why doesn't this match eGRID?
The average GGL for the entire countyr is 2.77%, but eGRID reports double this value: 5.3%

In [28]:

ba_disposition.sum()["total_energy_losses_mwh"] / (ba_disposition.sum()["total_disposition_mwh"] - ba_disposition.sum()["sales_for_resale_mwh"])

C:\Users\Greg\AppData\Local\Temp\ipykernel_16404\114092477.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  ba_disposition.sum()["total_energy_losses_mwh"] / (ba_disposition.sum()["total_disposition_mwh"] - ba_disposition.sum()["sales_for_resale_mwh"])


0.05410528777430053

In [32]:
hi_bas = ["HECO","HIMS"]
ba_disposition[ba_disposition["ba_code"].isin(hi_bas)]

ba_code  sales_for_resale_mwh  total_energy_losses_mwh  \
21    HECO                   0.0                 276888.0   
22    HIMS                   0.0                 129672.0   

    total_disposition_mwh  grid_gross_loss  
21           6.651997e+06         0.041625  
22           2.500285e+06         0.051863

In [33]:
ba_disposition[ba_disposition["ba_code"] == "ERCO"]

ba_code  sales_for_resale_mwh  total_energy_losses_mwh  \
15    ERCO          2.637507e+08             1.309243e+07   

    total_disposition_mwh  grid_gross_loss  
15           6.344030e+08         0.035323

In [34]:
ak_bas = ["AMPL","AKMS","CEA"]
ba_disposition[ba_disposition["ba_code"].isin(ak_bas)]

ba_code  sales_for_resale_mwh  total_energy_losses_mwh  \
2    AKMS               54560.0                 156587.0   
3    AMPL              101292.0                      0.0   
8     CEA               64183.0                 100514.0   

   total_disposition_mwh  grid_gross_loss  
2              3605061.0         0.044103  
3               834120.0         0.000000  
8              1382154.0         0.076264